pkgs/development/python-modules/sip/4.x.nix

```diff
@@ -39,11 +45,6 @@ buildPythonPackage rec {
 
   enableParallelBuilding = true;
 
-  pythonImportsCheck = [
```

Comment:
why remove the importsCheck?

Reply:
The check now segfaults on my system. I am unsure why, as when I manually test it, it works fine.

Reply:
I did some more investigating, and this appears to be an upstream bug that is present even in the sip4 version from ubuntu 24.04. python will segfault on exit after importing sip. Here is that reproduced on my fresh ubuntu install: 
![image](https://github.com/user-attachments/assets/0d50595d-3871-491d-ae0a-a3caba57117e)

Perhaps this is related: https://www.riverbankcomputing.com/pipermail/pyqt/2023-January/045094.html 

In my opinion, we this is still better than no support for python3.12. The package works perfectly fine until exiting, so at least dependent software will function. Perhaps in another PR we can fix this bug, but honestly, I would hope that sip4 will become much less relevant soon anyways.

@gador let me know if you disagree.

Reply:
Following on that email trail, it seems like an upstream bug, which hasn't been resolved. In this case, I agree, we should remove the import check for `sip` for python >=3.12 (as you have done). Maybe we should add a comment explaining this, though. (A link to either the email list, or this comment/PR should do fine)

Reply:
comment added in for the pythonImportsCheck change